pkg/notify/webhook/webhook.go

```diff
@@ -0,0 +1,87 @@
+package webhook
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/ovh/utask/pkg/notify"
+)
+
+const (
+	// Type represents Webhook as notify backend
+	Type string = "webhook"
+)
+
+// NotificationSender is a notify.NotificationSender implementation
+// capable of sending notifications to a webhook
+type NotificationSender struct {
+	webhookURL string
+	username   string
+	password   string
+	headers    map[string]string
+	httpClient *http.Client
+}
+
+// NewWebhookNotificationSender instantiates a NotificationSender
+func NewWebhookNotificationSender(webhookURL, username, password string, headers map[string]string) *NotificationSender {
+	return &NotificationSender{
+		webhookURL: webhookURL,
+		username:   username,
+		password:   password,
+		headers:    headers,
+		httpClient: &http.Client{Timeout: 10 * time.Second},
+	}
+}
+
+// Send is the implementation for triggering a webhook to send the notification
+func (w *NotificationSender) Send(m *notify.Message, name string) {
```

Comment:
This should probably return an error and delegate the error handling to the caller.
It would require an update of the `notify.NotificationSender` interface.

Reply:
Let's consider that notification doesn't have any SLA for delivering.
We can address this later.